pkg/apis/metering/v1/util/report_util.go

```diff
@@ -71,6 +71,11 @@ func GetReportCondition(status metering.ReportStatus, condType metering.ReportCo
 // SetReportCondition updates the report to include the provided condition. If the condition that
 // we are about to add already exists and has the same status and reason then we are not going to update.
 func SetReportCondition(status *metering.ReportStatus, condition metering.ReportCondition) {
+	// TODO we accept a pointer but do not account for nil.  This seems like a programmatic error, should we
+	// throw an error rather than return since the caller is definitely expecting an action to happen?
```

Comment:
Indeed, we should throw and the several call sites should also throw on hearing about the error it looks like. I'll need to look more myself to see if we have existing error handling code that could deal with this at some point up the chain.

It looks like 2 of the 3 current call sites can't send a null pointer, but one is just passing along `report.status` from a `report` it got, so it could, this is in `updateReportStatus()`. It does look like all the callers of `updateReportStatus` are prepared to hear about errors, so I that makes me even more confident about the above error throwing.

Reply:
Thanks @bentito.  If you think we should throw here then I'll take a look at what refactoring that would cause and add it in.